opentelemetry-example-app/tests/test_flask_example.py

```diff
@@ -1,14 +1,57 @@
 import unittest
+from unittest import mock
+
+import requests
+from werkzeug.test import Client
+from werkzeug.wrappers import BaseResponse
 
 import opentelemetry_example_app.flask_example as flask_example
+from opentelemetry.sdk import trace
+from opentelemetry.sdk.context.propagation import b3_format
 
 
 class TestFlaskExample(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
         cls.app = flask_example.app
 
+    def setUp(self):
+        mocked_response = requests.models.Response()
+        mocked_response.status_code = 200
+        mocked_response.reason = "Roger that!"
+        self.send_patcher = mock.patch.object(
+            requests.Session,
+            "send",
+            autospec=True,
+            spec_set=True,
+            return_value=mocked_response,
+        )
+        self.send = self.send_patcher.start()
+
+    def tearDown(self):
+        self.send_patcher.stop()
+
     def test_full_path(self):
-        with self.app.test_client() as client:
-            response = client.get("/")
-            assert response.data.decode() == "hello"
+        trace_id = trace.generate_trace_id()
+        # We need to use the Werkzeug test app because
+        # The headers are injected at the wsgi layer.
+        # The flask test app will not include these, and
+        # result in the values not propagated.
+        client = Client(self.app.wsgi_app, BaseResponse)
+        # emulate b3 headers
+        client.get(
+            "/",
+            headers={
```

Comment:
We could use the B3 formatter's inject directly on the dict here.

Reply:
that might be a little tricky: it would require that I have direct access to the SpanContext for the app, which may not occur in situations where the app lives in a different thread or context than the test code itself.

I feel like this is a more thorough test of the defined behavior, although I definitely see the merit of not effectively redefining the b3 interface.